javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -92,8 +98,10 @@ public static void clear() {
      * @return an {@link AnnotationMirror} of type {@code} name
      */
     public static AnnotationMirror fromName(Elements elements, CharSequence name) {
-        if (annotationsFromNames.containsKey(name)) {
-            return annotationsFromNames.get(name);
+        synchronized (annotationsFromNames) {
```

Comment:
Would it be shorter and still thread safe to do:

```
AnnotationMirror res = annotationsFromNames.get(name);
if (res != null) {
  return res;
}
...
```

Instead of checking and then getting the value, why not just get the value and see whether we got something?